src/training/flava_eval/glue2.py

```diff
@@ -0,0 +1,324 @@
+            raise ImportError("Please install HF datasets: pip install datasets")
+        
+        self.dataset = load_dataset("glue", task, split=split)
+        self.label_key = label_key
+        self.text_key = text_key # list of strings
+        self.length = len(self.dataset)
+        self.tokenize = tokenizer
+        self.task = task
+        self.separator_token = separator_token
+
+    def __len__(self):
+        return self.length
+
+    def __getitem__(self, idx):
+        if self.task in MULTI_PROMPT_TASKS:
+            item = self.dataset[idx]
+            text1 = item[self.text_key[0]]
+            text2 = item[self.text_key[1]]
+            label = item[self.label_key]
+            return {
+                'text': self.tokenize([text1 + self.separator_token + text2])[0],
+                'label': label
+            }
+        else:
+            item = self.dataset[idx]
+            text = item[self.text_key[0]]
+            label = item[self.label_key]
+            return {
+                'text': self.tokenize([text])[0],
+                'label': label
+            }
+
+
+class TextClassifier(nn.Module):
+    def __init__(self, encoder, embed_dim, num_labels):
+        super().__init__()
+
+        self.encoder = encoder
+        self.logits_proj = nn.Linear(embed_dim, num_labels)
+
+    def forward(self, text):
+        text_features = self.encoder.encode_text(text)
+        logits = self.logits_proj(text_features)
+        return logits
+
+
+def get_task_metric(task_name):
+    if task_name == "cola":
+        metric = evaluate.load("glue", "stsb")
+    else:
+        metric = evaluate.load("glue", task_name)
+    return metric
+
+
+def get_task_dataloaders(args):
+    tokenizer = get_tokenizer(args.model)
+    task_name = args.task_name
+
+    dataloaders = {}
+    for split_name in ["train", args.validation_key, args.test_key]:
+        dataset = GLUEDataset(
+            task_name,
+            split_name,
+            text_key=TEXT_KEYS[task_name],
+            label_key="label",
+            tokenizer=tokenizer,
+            separator_token=args.separator_token
+        )
+        dataloader = DataLoader(
+            dataset,
+            batch_size=args.batch_size,
+            shuffle=True,
+            num_workers=args.workers,
+            pin_memory=True,
+            drop_last=True,
+        )
+        dataloaders[split_name] = dataloader
+
+    return dataloaders
+
+
+def compute_metrics(model, dataloader, device, args):
+    model.eval()
+    metric = get_task_metric(args.task_name)
+    val_loss = 0
+    samples_seen = 0
+    for batch in dataloader:
+        with torch.no_grad():
+            text = batch["text"].to(device)
+            label = batch["label"].to(device)
+            samples_seen += text.shape[0]
+            logits = model(text)
+            logits = logits.view(-1)
+            label = label.view(-1).float()
+            predictions = torch.sigmoid(logits) > 0.5
```

Comment:
Not all the tasks are binary classification, right? Some are regression and some are multi-class, what do you do here?